### PR TITLE
Feat/test computation engines

### PR DESCRIPTION
**Pull request by Baboo7**
## Proposed changes

---

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

---

What types of changes does your code introduce to ogre ?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe):

## Checklist

---

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

offseason_ogre/blob/master/CONTRIBUTING.md) doc

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation and modify the README.md (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] If the feature changes current behavior, reasons why your solution is better.

## What is the current behavior?

---

-
-
-

## What is the new behavior?

---

-
-
-

## Further comments

---

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

## Where has this been tested?

---

Docker version: …
Operating System: …
node version: …
postgresql version: …

### CODE REVIEW

File: packages/client/src/modules/play/utils/consumption.spec.ts [hunk 1]
```diff
@@ -0,0 +1,631 @@
...
+          consumption: [
+            { type: "grey", name: "greyOther", value: 10 },
+            { type: "grey", name: "greyTransport", value: 10 },
+          ],
+          performedActions: [availableActions.LOCAL_CONSUMPTION],
+        },
+        expected: {
+          consumptionData: [
+            { type: "grey", name: "greyOther", value: 4.6 },
+            { type: "grey", name: "greyTransport", value: 8.8 },
```
Comment (louis-sanna-dev) [inline packages/client/src/modules/play/utils/consumption.spec.ts:70]: @Baboo7 c'est vraiment top que tu écrives des tests, mais ils sont fragiles car on va devoir les changer à chaque changement du modèle. Que penses-tu de tester une inégalité du type `newGreyOther < oldGreyOther` ?

Comment (Baboo7) [reply packages/client/src/modules/play/utils/consumption.spec.ts:70]: @louis-sanna-eki je pense que ça vaut le coup de travailler en valeur exacte pour mieux controler les calculs, notamment l'ordre dans lequel les actions sont traitées. Qu'est ce que vous en pensez @remiriv @Vladou73 ?

Comment (louis-sanna-dev) [reply packages/client/src/modules/play/utils/consumption.spec.ts:70]: @Baboo7 ok pour moi.

**Review by louis-sanna-dev — APPROVED** [review]

File: packages/client/src/modules/play/utils/consumption.spec.ts (hunk 1 above)
Comment (Baboo7) [reply packages/client/src/modules/play/utils/consumption.spec.ts:70]: Ça marche 👌

Comment (remiriv) [reply packages/client/src/modules/play/utils/consumption.spec.ts:70]: @Baboo7 j'avais pas vu mais oui je trouve ça plus pratique d'avoir des vrais tests de valeurs 
après peut être que tout dans un seul fichier c'est un peu dense si on doit modifier les tests de temps en temps, à voir pour les prochains si on fait pas des fichiers plus petits par thème de tests

**Pull request merged**